firmware/console/binary/tunerstudio.cpp

```diff
@@ -751,85 +720,88 @@ int tunerStudioHandleCrcCommand(ts_channel_s *tsChannel, char *data, int incomin
 	char command = data[0];
 	data++;
 
-	// Output command first since it's popular
-	if (command == TS_OUTPUT_COMMAND) {
-		uint16_t offset = *(uint16_t *) (data);
-		uint16_t count = *(uint16_t *) (data + 2);
-		handleOutputChannelsCommand(tsChannel, TS_CRC, offset, count);
-	} else if (command == TS_HELLO_COMMAND) {
+	const uint16_t* data16 = reinterpret_cast<uint16_t*>(data);
+
+	switch(command)
```

Comment:
Changing to a switch has a few benefits:  Not only does it make it clearer to the reader that this is a "pick one" scenario, it also gives the compiler more leeway about optimization.  `if else` statements are required to be tested in order, but `switch` isn't.  This means the compiler can do smart things like replace it with a jump table (which it does with `-O2` enabled), so that this is an O(1) lookup instead of an O(n) lookup.

Reply:
(looking at `rusefi.list`, it uses the super cool `tbh` instruction: http://www.keil.com/support/man/docs/armasm/armasm_dom1361289912399.htm)